packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -0,0 +1,596 @@
+        tokenId: tokenId,
+        liquidity: positionBefore.liquidity,
+        amount0Min: amount0Min,
+        amount1Min: amount1Min,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+      })
+      const wPowerPerpAmountInLP = (isWethToken0) ? amount1 : amount0;
+      const wethAmountInLP = (isWethToken0) ? amount0 : amount1;
+      const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 420, true)
+      const depositorEthBalanceBefore = await provider.getBalance(depositor.address)
+      const slippage = BigNumber.from(3).mul(BigNumber.from(10).pow(16))
+      const minEthPerWPowerPerp = squeethPrice.mul(one.sub(slippage)).div(one);
+
+      await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address);
+      await (positionManager as INonfungiblePositionManager).connect(depositor).approve(controllerHelper.address, tokenId); 
+      await controllerHelper.connect(depositor).closeShortWithUserNft({
+        vaultId, 
+        tokenId,
+        liquidityPercentage: BigNumber.from(1).mul(BigNumber.from(10).pow(18)),
+        wPowerPerpAmountToBurn: mintWSqueethAmount, 
+        collateralToWithdraw: vaultBefore.collateralAmount, 
+        minEthPerWPowerPerp, 
+        amount0Min: BigNumber.from(0), 
+        amount1Min:BigNumber.from(0)
+      })
+
+      const positionAfter = await (positionManager as INonfungiblePositionManager).positions(tokenId);
+      const vaultAfter = await controller.vaults(vaultId);
+      const depositorEthBalanceAfter = await provider.getBalance(depositor.address)
+
+      expect(positionAfter.liquidity.eq(BigNumber.from(0))).to.be.true
+      expect(vaultAfter.shortAmount.eq(BigNumber.from(0))).to.be.true
+      expect(vaultAfter.collateralAmount.eq(BigNumber.from(0))).to.be.true
+
+      if(wPowerPerpAmountInLP.lt(mintWSqueethAmount)) {
+        const ethToBuySqueeth = (mintWSqueethAmount.sub(wPowerPerpAmountInLP)).mul(squeethPrice).div(one); 
+        const remainingETHFromLp = wethAmountInLP.sub(ethToBuySqueeth);
+
+        expect(Number(depositorEthBalanceAfter.sub(depositorEthBalanceBefore).sub(vaultBefore.collateralAmount.add(remainingETHFromLp)).div(one).toString()) <= 0.01).to.be.true
+      }
+      else if (wPowerPerpAmountInLP.gt(mintWSqueethAmount)) {
+        const wPowerPerpAmountToSell = wPowerPerpAmountInLP.sub(mintWSqueethAmount);
+        const ethToGet = wPowerPerpAmountToSell.mul(squeethPrice).div(one);
+
+        expect(Number(depositorEthBalanceAfter.sub(depositorEthBalanceBefore).sub(vaultBefore.collateralAmount.add(ethToGet).add(wethAmountInLP)).div(one).toString()) <= 0.01).to.be.true
+      }
+    })
+  })
+
+  describe("Close second position with user wallet NFT from 1st short: remove 50% liquidity LP wPowerPerp amount is more than vault short amount", async () => {
```

Comment:
this one should be something like 40% to be less than the short amount (maybe it is slightly w rounding, but could be more explicity)?

vault 1 = 20 oSQTH
vault 2 = 10 oSQTH
LP = 20 oSQTH (50% = 10)

Reply:
I'm trying to have liquidity that have more wPowerPerp that short amount, so yeah here IMO 60%